extensions/iam/oauth2/oauth2-core/src/main/java/org/eclipse/dataspaceconnector/iam/oauth2/core/Oauth2Extension.java

```diff
@@ -76,14 +80,29 @@
     private long keyRefreshInterval;
 
     private ScheduledExecutorService executorService;
+
     @Inject
     private OkHttpClient okHttpClient;
 
-    private static Supplier<JWSSigner> createRsaPrivateKeySupplier(Oauth2Configuration configuration) {
+    private static Supplier<JWSSigner> createPrivateKeySupplier(Oauth2Configuration configuration) {
```

Comment:
I know that this "supplier pattern" was already in place, but I think is overengineered, since the supplier is called in the constructor of `Oauth2ServiceImpl`, I would take this opportunity to "inline" it: create the signer directly in the extension initialization and passing it to the service only if it's instantiated correctly

Reply:
Done.